### PR TITLE
crush: verify weights is influenced by the number of replicas

### PR DESCRIPTION
**Pull request by dachary**
Refs: http://tracker.ceph.com/issues/15653

Signed-off-by: Adam C. Emerson <aemerson@redhat.com>
Signed-off-by: Loic Dachary <loic@dachary.org>

### CODE REVIEW

**Comment by dachary** [comment]
Borrowed from Adam's pull request https://github.com/ceph/ceph/pull/10218/files#diff-2dc54bd8be542969c190bd3c890d14d7R3

File: src/test/crush/crush_weights.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    step emit
+}
+EOF
+
+old=($(echo "$cm" | crushtool -c /dev/fd/0 --test --show-utilization \
+                              --min-x 1 --max-x 1000000 --num-rep 3 | \
+  grep "device \(0\|4\)" | sed -e 's/^.*stored : \([0-9]\+\).*$/\1/'))
+
+if test $(echo "scale=5; (10 - ${old[0]}/${old[1]}) < .75" | bc) = 1; then
+    echo Untweaked weights better distributed than they should be. 1>&2
```
Comment (liewegas) [inline src/test/crush/crush_weights.sh]: this message doesn't make sense outside of adam's proposed fix.  otherwise lgtm!

Comment (liewegas) [inline src/test/crush/crush_weights.sh]: also, perhaps the test should do the same thing with num-rep 1 and verify the ratio is in the expected range

**Comment by dachary** [comment]
Addressed sage comments (fixed the error message and added a test for replica 1), chmod +x and repushed.

**Pull request merged**